src/Core/Resolvers/SqlPaginationUtil.cs

```diff
@@ -62,26 +62,33 @@ private static JsonObject CreatePaginationConnection(JsonElement root, Paginatio
                 root = document.RootElement.Clone();
             }
 
-            IEnumerable<JsonElement> rootEnumerated = root.EnumerateArray();
+            // If the request includes either hasNextPage or endCursor then to correctly return those
+            // values we need to determine the correct pagination logic
+            bool paginationScenario = paginationMetadata.RequestedHasNextPage || paginationMetadata.RequestedEndCursor;
 
+            IEnumerable<JsonElement> rootEnumerated = root.EnumerateArray();
+            int returnedElemNo = rootEnumerated.Count();
```

Comment:
nit: since touching this area of code, perhaps `returnedElemNo` can have a more descriptive name like `returnedElementCount`